src/phoenix/experimental/evals/functions/classify.py

```diff
@@ -130,8 +218,12 @@ def llm_classify(
             except json.JSONDecodeError:
                 unrailed_label = response
                 explanation = None
-        labels.append(_snap_to_rail(unrailed_label, rails, verbose=verbose))
-        explanations.append(explanation)
+        return _snap_to_rail(unrailed_label, rails, verbose=verbose), explanation
+
+    executor = AsyncExecutor(_classify_prompt, tqdm_bar_format=tqdm_bar_format)
+    results = executor.run(prompts.tolist())
```

Comment:
I added the ability to recover from a KeyboardInterrupt in my pr #1755 - I think it's actually a pretty necessary feature, is it possible to parameterize the results so that the partial results can be recovered?

Reply:
I'm quite sure it's possible to recover this behavior in several ways in this refactor, but let's sync up to make sure it'll work out? Alternatively I can try and adapt/merge your strategy directly in this PR?